planner/core/logical_plans.go

```diff
@@ -586,6 +586,10 @@ type DataSource struct {
 	// 1. use `inside insert`, `update`, `delete` or `select for update` statement
 	// 2. isolation level is RC
 	isForUpdateRead bool
+
+	// contain unique index and the first field is tidb_shard(),
+	// such as (tidb_shard(a), a ...), the fields are more than 2
+	containExprPrefixUk bool
```

Comment:
This should be the property of the meta data of the table instead of the logical plan?

Reply:
Yes, I did it as you said at first. But we know that tiancaimao will resolve the hot index fully in the future, and our team think that I should do this as simple as possible, so I don't import a new syntax or change the process of ddl.
Please see the review comment above by cfzjywxk.